src/app/ipv/middleware.js

```diff
@@ -144,11 +147,21 @@ async function handleBackendResponse(req, res, backendResponse) {
     req.session.context = backendResponse?.context;
     req.session.currentPageStatusCode = backendResponse?.statusCode;
 
-    return await saveSessionAndRedirect(
-      req,
-      res,
-      getIpvPagePath(req.session.currentPage),
-    );
+    await req.session.save(function (err) {
+      if (err) {
+        logError(req, err, "Error saving session");
+        throw err;
+      }
+    });
+
+    // Special case handling for "identify-device". This is used by core-back to signal that we need to
+    // check the user's device and send back the relevant "appTriage" event.
+    if (backendResponse.page === PAGES.IDENTIFY_DEVICE) {
```

Comment:
It feels a bit weird this being a page response when it's not actually a page 🤔 
Why is it we need this for the CoI scenarios? I think originally we just always sniffed the device to get the right appTriage event without an additional call to core-back

Reply:
It is weird and I don't hugely like it, but it looks like our options are:
1. Do this
2. Introduce a new type of response from core-back for just this case
3. Have a huge number of journey map states for handling the name change routes as they go directly from a backend processing step to DCMAW at the moment, so we'd need turn each existing backend step into 3 to then go to the correct appTriage page afterwards.

Reply:
We agreed to go with this for now, with a tech debt ticket raised to consider adding a new response type - [PYIC-7534](https://govukverify.atlassian.net/browse/PYIC-7534)